src/Grpc.Net.Client/Balancer/SubchannelsLoadBalancer.cs

```diff
@@ -144,8 +145,11 @@ public override void UpdateChannelState(ChannelState state)
                 // The new subchannel address has the same endpoint so the connection isn't impacted.
                 if (!BalancerAddressEqualityComparer.Instance.Equals(address, newOrCurrentSubchannel.Address))
                 {
+                    newOrCurrentSubchannel = new AddressSubchannel(
```

Comment:
Another way to express this might be to expose a `WithAddress` method on `AddressSubchannel` that returns a new object differing only in the address.